File: AZ-204_Implement_Azure_App_Services_Web_Apps/code/resource_mgmnt.py
from azure.identity import DefaultAzureCredential
from azure.mgmt.web import WebSiteManagementClient

# Authenticate using default credentials
credential = DefaultAzureCredential()

# Create a WebSiteManagementClient
client = WebSiteManagementClient(credential, subscription_id)

# Define the resource group and app service name
resource_group = "my-resource-group"
app_service_name = "my-app-service"

# Create the app service
app_service = client.web_apps.create_or_update(
    resource_group,
    app_service_name,
    {
        "location": "westus",
        "server_farm_id": "/subscriptions/{subscription_id}/resourceGroups/{resource_group}/providers/Microsoft.Web/serverfarms/{server_farm_name}"
    }
)

print("App Service created successfully!")

# List app services
app_services = client.web_apps.list_by_resource_group(resource_group)
for app_service in app_services:
    print(app_service.name)

# Update app service
app_service = client.web_apps.get(resource_group, app_service_name)
app_service.https_only = True
app_service.tags = {"key1": "value1", "key2": "value2"}
client.web_apps.create_or_update(resource_group, app_service_name, app_service)
print("App Service updated successfully!")

# Delete app service
client.web_apps.delete(resource_group, app_service_name)
print("App Service deleted successfully!")

File: AZ-204_Implement_Azure_App_Services_Web_Apps/code/resource_mgmnt.cs
using Azure.Identity;
using Azure.ResourceManager.AppService;
using Azure.ResourceManager.AppService.Models;
using Azure.ResourceManager.Resources;

class Program
{
    static async Task Main(string[] args)
    {
        string subscriptionId = "<your-subscription-id>";
        string resourceGroupName = "<your-resource-group-name>";
        string appName = "<your-app-name>";
        string location = "<your-location>";

        var credential = new DefaultAzureCredential();
        var resourceClient = new ResourcesManagementClient(subscriptionId, credential);
        var appServiceClient = new AppServiceManagementClient(subscriptionId, credential);

        // Create resource group
        await resourceClient.ResourceGroups.CreateOrUpdateAsync(resourceGroupName, new ResourceGroup(location));

        // Create app service plan
        var appServicePlan = new AppServicePlan(location, "S1");
        await appServiceClient.AppServicePlans.CreateOrUpdateAsync(resourceGroupName, appName, appServicePlan);

        // Create app service
        var appService = new WebApp(location);
        await appServiceClient.WebApps.CreateOrUpdateAsync(resourceGroupName, appName, appService);

        // Update app service properties
        var updatedAppService = await appServiceClient.WebApps.GetAsync(resourceGroupName, appName);
        updatedAppService.HttpsOnly = true;
        updatedAppService.Tags.Add("Environment", "Production");
        updatedAppService.Tags.Add("Owner", "John Doe");
        await appServiceClient.WebApps.UpdateAsync(resourceGroupName, appName, updatedAppService);

        // List app services
        var appServices = await appServiceClient.WebApps.ListByResourceGroupAsync(resourceGroupName);
        foreach (var app in appServices)
        {
            Console.WriteLine($"App Service Name: {app.Name}");
            Console.WriteLine($"App Service URL: {app.DefaultHostName}");
            Console.WriteLine();
        }

        // Delete app service
        await appServiceClient.WebApps.StartDeleteAsync(resourceGroupName, appName);
    }
}